packages/driver/cypress/e2e/e2e/origin/snapshots.cy.ts

```diff
@@ -2,107 +2,153 @@
+      cy.origin('http://www.foobar.com:3500', () => {
+        // need to set isInteractive in the spec bridge in order to take xhr snapshots in run mode, similar to how isInteractive is set within support/defaults.js
+        // @ts-ignore
+        Cypress.config('isInteractive', true)
+        cy.visit('http://www.foobar.com:3500/fixtures/xhr-fetch-requests.html')
+        cy.get(`[data-cy="assertion-header"]`).should('exist')
+        cy.wait('@fooBarBaz')
+      })
+
+      cy.shouldWithTimeout(() => {
+        const xhrLogFromSecondaryOrigin = findLog(logs, 'xhr', 'http://localhost:3500/foo.bar.baz.json')?.get()
 
-      expect(xhrLogFromSecondaryOrigin).to.not.be.undefined
+        expect(xhrLogFromSecondaryOrigin).to.not.be.undefined
 
-      const snapshots = xhrLogFromSecondaryOrigin.snapshots.map((snapshot) => snapshot.body.get()[0])
+        const snapshots = xhrLogFromSecondaryOrigin.snapshots.map((snapshot) => snapshot.body.get()[0])
 
-      snapshots.forEach((snapshot) => {
-        expect(snapshot.querySelector(`[data-cy="assertion-header"]`)).to.be.null
+        expect(snapshots.length).to.equal(2)
+
+        // TODO: Since we have two events, one of them does not have a request snapshot
+
+        expect(snapshots[1].querySelector(`[data-cy="assertion-header"]`)).to.have.property('innerText').that.equals('Making XHR and Fetch Requests behind the scenes if fireOnload is true!')
+      })
+    })
+
+    // TODO: fix flaky test https://github.com/cypress-io/cypress/issues/23437
+    it.skip('verifies fetch requests made while a secondary origin is active eventually update with snapshots of the secondary origin', () => {
+      cy.origin('http://www.foobar.com:3500', () => {
+        // need to set isInteractive in the spec bridge in order to take xhr snapshots in run mode, similar to how isInteractive is set within support/defaults.js
+        // @ts-ignore
+        Cypress.config('isInteractive', true)
+        cy.visit('http://www.foobar.com:3500/fixtures/xhr-fetch-requests.html')
+        cy.get(`[data-cy="assertion-header"]`).should('exist')
+        cy.wait('@fooBarBaz')
       })
 
-      done()
+      cy.shouldWithTimeout(() => {
+        const xhrLogFromSecondaryOrigin = findLog(logs, 'fetch', 'http://localhost:3500/foo.bar.baz.json')?.get()
+
+        expect(xhrLogFromSecondaryOrigin).to.not.be.undefined
+
+        const snapshots = xhrLogFromSecondaryOrigin.snapshots.map((snapshot) => snapshot.body.get()[0])
+
+        snapshots.forEach((snapshot) => {
+          expect(snapshot.querySelector(`[data-cy="assertion-header"]`)).to.have.property('innerText').that.equals('Making XHR and Fetch Requests behind the scenes if fireOnload is true!')
+        })
+      })
     })
 
-    cy.visit('http://www.foobar.com:3500/fixtures/xhr-fetch-requests.html')
+    it('Does not take snapshots of XHR/fetch requests from secondary origin if the wrong origin is / origin mismatch, but instead the primary origin (existing behavior)', {
```

Comment:
I read this 2x and am a little confused

Reply:
I think this is missing the word `visited` in the name. I can add it.

Reply:
addressed in 1d4323475b